src/eth.ts

```diff
@@ -9,19 +9,24 @@ const log = factory.getLogger(formatFilename(__filename));
 export async function createProvider(
   ethNetworkUrl: string
 ): Promise<providers.Web3Provider> {
-  if (ethNetworkUrl.includes('infura')) {
-    const networkPrefix = ethNetworkUrl.split('infura')[0];
-    if (process && process.env) {
-      const { INFURA_API_KEY } = process.env;
-      if (!INFURA_API_KEY) {
-        throw new Error('no infura key found!');
-      }
-      ethNetworkUrl = `${networkPrefix}infura.io/ws/v3/${INFURA_API_KEY}`;
-
-      let res;
-      let data;
-      try {
-        res = await fetch(`https://mainnet.infura.io/v3/${INFURA_API_KEY}`, {
+  if (!ethNetworkUrl.includes('alchemy'))
```

Comment:
Heads up that this wont work as a drop-in replacement, as each of the chains receive their URL from the ChainNode stored on the database. We should, for the time being, rewrite Infura URLs to Alchemy (I may do this myself and then merge).